src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -121,20 +124,26 @@ public void configure() throws Exception {
                         .throwException(org.apache.commons.httpclient.HttpException.class, "Unsuccessful response from Insights Upload Service")
                 .end();
 
-        from("kafka:" + kafkaHost + "?topic={{insights.kafka.upload.topic}}&brokers=" + kafkaHost + "&autoOffsetReset=latest&autoCommitEnable=true")
+        from("kafka:" + kafkaHost + "?topic={{insights.kafka.upload.topic}}&brokers=" + kafkaHost + "&autoOffsetReset=earliest&autoCommitEnable=true")
                 .routeId("kafka-upload-message")
                 .unmarshal().json(JsonLibrary.Jackson, FilePersistedNotification.class)
                 .filter(simple("'{{insights.service}}' == ${body.getService}"))
                 .to("direct:download-file");
 
         from("direct:download-file")
                 .routeId("download-file")
-                .setHeader("Exchange.HTTP_URI", simple("${body.url}"))
+                .setHeader("tempHTTP_URI", simple("${body.url}"))
+                .process(e -> {
+                    String headerURI = "tempHTTP_URI";
+                    e.getIn().setHeader(headerURI, e.getIn().getHeader(headerURI, String.class).replace("minio:9000", minio_host));
+                })
+                .setHeader("Host", constant("minio:9000"))
+                .setHeader("Exchange.HTTP_URI", simple("${header.tempHTTP_URI}"))
```

Comment:
I think we are mixing Test code with the original code. What about something like:
```
from("direct:download-file")
                .routeId("download-file")
                .setHeader("Exchange.HTTP_URI", simple("${body.url}"))
// Here we can write the code for the testing environment
                .choice()
                    .when(exchange -> ((String) exchange.getIn().getBody()).contains("minio:9000"))
                        .process(e -> e.getIn().setHeader("Exchange.HTTP_URI",  minio_host))
                        .setHeader("Host", constant("minio:9000"))
                .end()

                .convertBodyTo(FilePersistedNotification.class)
```
I'm proposing this just to be sure which code will run in Test environment

Reply:
Yep, I'm already changing that ... in another way, removing that piece of code from the final route.
Thank you